examples/tic-tac-toe/ui/src/hooks/useGameQuery.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { UseQueryResult } from '@tanstack/react-query';
+import { useNetworkVariable } from 'config';
+import { useObjectQuery, UseObjectQueryResponse } from 'hooks/useObjectQuery';
+
+/** Variants of the tic-tac-toe protocol */
+export type Kind = 'shared' | 'owned';
+
+/** Marks on the game baord */
+export enum Mark {
+	/** No mark */
+	_ = 0,
+	/** Marked by player X */
+	X,
+	/** Marked by player O */
+	O,
+}
+
+/** State of the game, deserialized from its Move Object */
+export type Game = {
+	/** The game's Object ID */
+	id: string;
+
+	/** Whether it's a 'shared' or an 'owned' game */
+	kind: Kind;
+
+	/** Current state of the game board, 9 marks in row-major order. */
+	board: Mark[];
+
+	/** Number of turns played so far. */
+	turn: number;
+
+	/** Address of the player controlling X */
+	x: string;
+
+	/** Address of the player controlling O */
+	o: string;
+
+	/** The multi-sig public-key of the game admin */
+	admin?: number[];
+};
+
+export type UseGameQueryResult = UseQueryResult<Game, Error>;
+export type InvalidateGameQuery = () => void;
+
+/** Refetch games every 5 seconds */
+const REFETCH_INTERVAL = 5000;
```

Comment:
I like option (1) a fair bit, but it doesn't play too nicely with how `dapp-kit` currently handles keys (you can only supply a key suffix, not a prefix) -- I wonder if there's a way we can tweak the way `dapp-kit` works to allow these kind of patterns, cc @hayes-mysten . (2) seems less ideal to me, because it sounds like we would switch things around to make refetching the default, and then have to manually turn it off in places, which would work for our case here, but I'm not sure how generalisable that is.

What I was originally hoping was that there was a way to describe the data dependency, so that `react-query` could detect that there was a change in the query input and therefore re-run it, in which case we could treat our game as a redundant input to all the other queries, and use that mechanism to trigger the refetches -- is that feasible at all?

Reply:
Going to defer to @williamrobertson13 here who has some more react/front-end expertise

Reply:
this is definitely something we should have a good (and well documented) solution for